src/js/mod.rs

```diff
@@ -48,26 +49,37 @@ pub(crate) mod internal {
     }
 
     #[repr(C)]
-    pub struct FunctionKernel<T: Value>(pub fn(Call) -> JsResult<T>);
+    pub struct FunctionKernel<T: Value>(pub Box<FnMut(Call) -> JsResult<T>>);
```

Comment:
Might be worth having type parameter `F: FnMut(Call) -> JsResult<T>` to avoid dynamic dispatch.